File: src/java/linkedlists/HasCycles.java
package java.linkedlists;

public class HasCycles {
    public static class Node {
        int d;
        Node next;

        public Node(int d) {
            this.d = d;
        }
    }

    public static boolean hasCycles(Node head) {
        Node fast = head;
        Node slow = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;

            if (slow == fast) {
                return true;
            }
        }

        return false;
    }
}

File: src/main/kotlin/lists/HasCycles.kt
package lists

class HasCycles {
    class Node(val value: Int, var next: Node? = null)

    fun hasCycles(head: Node?): Boolean {
        var fast = head
        var slow = head

        while (fast != null && fast.next != null) {
            slow = slow?.next
            fast = fast.next?.next

            if (slow == fast) {
                return true
            }
        }

        return false
    }
}